### PR TITLE
[#642] Activate useJenkinsSnapshots profile in test

### PR DESCRIPTION
**Pull request by kthoms**
Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by kthoms** [comment]
https://ci.eclipse.org/xtext/view/Xtext-release/job/xtext-xtend/job/milestone_2.16.0.M3/
is green. That was the failing job.

**Comment by cdietrich** [comment]
but i still wonder if this is affected by the symptoms described in issue #608 
we still have the problems that the paths of the local maven repos are simply incorrect

File: org.eclipse.xtend.maven.plugin/pom.xml [hunk 1]
```diff
@@ -218,7 +218,7 @@
 				<artifactId>maven-surefire-plugin</artifactId>
 				<version>2.10</version>
 				<configuration>
-					<argLine>-Xmx1024m -DgradleMavenRepo=${gradleMavenRepo}</argLine>
```
Comment (cdietrich) [inline org.eclipse.xtend.maven.plugin/pom.xml]: non of the tests define this Profile nor is the the maven plugin parent a parent of the tests.
so i dont think this fixes anything.
it instead uses the local maven repo.
thus the maven repo und gradle and maven must be consitent

**Comment by kthoms** [comment]
Maybe something for @pngowda to analyze

**Comment by cdietrich** [comment]
am not sure. last time it took me a whole day of debugging. unfort. the ci.eclipse.org environment is a big crap of s..... it behaves totally differnt from typefox when it comes how jenkins behaves

File: org.eclipse.xtend.maven.plugin/pom.xml (hunk 1 above)
Comment (kthoms) [reply org.eclipse.xtend.maven.plugin/pom.xml]: The parent of `org.eclipse.xtext.maven.plugin` is `org.eclipse.xtend.maven.parent` and that does declare the `useJenkinsSnapshots`

**Comment by cdietrich** [comment]
and: we have no branches to test against
=> we would have to try to build a rc1 branch.
maybe we should do this anyway to test version adaption scripts

**Comment by cdietrich** [comment]
no the tests that fail they dont use any of these parent poms

File: org.eclipse.xtend.maven.plugin/pom.xml (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtend.maven.plugin/pom.xml]: no the tests dont use that.
they use a rabitray shared maven repo
and the settings.xml crap at eclipse changes it

Comment (kthoms) [reply org.eclipse.xtend.maven.plugin/pom.xml]: Not sure. Trying to access Maven Central usually indicates that the profile was not activated.

Comment (cdietrich) [reply org.eclipse.xtend.maven.plugin/pom.xml]: on your branch / M3 no tests ran at all

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Picked up JAVA_TOOL_OPTIONS: -Xmx2G -Djava.io.tmpdir=/tmp/genie.xtext
Unrecognized option: -PuseJenkinsSnapshots
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.

Comment (kthoms) [reply org.eclipse.xtend.maven.plugin/pom.xml]: Oh :/

**Pull request closed without merging**